GVFS/GVFS.Common/Git/Sha1Id.cs

```diff
@@ -89,6 +90,11 @@ public static bool TryParse(string sha, out Sha1Id sha1, out string error)
             }
         }
 
+        public static string ShaStringFromBuffer(byte[] shaBuffer)
+        {
+            return new string(shaBuffer.SelectMany(b => new[] { GetHexValue(b / 16),  GetHexValue(b % 16) }).ToArray());

```

Comment:
Is this run in a tight loop? This use of Linq including a `new [] {}` in the middle of your loop is likely to be inefficient:

1. You create and destroy a lot of string arrays with the `new [] {}` in the middle.
2. `.ToArray()` is notoriously bad, since it cannot guess the size of the array at the start, so dynamically sizes a few arrays before truncating at the end. However, we know the length that `shaBuffer` should be.
3. Even the `.SelectMany()` is instantiating an enumerator that is unnecessary.

I think it is worth rewriting this method using [`StringBuilder` with a specified capacity](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.-ctor?view=netframework-4.8#System_Text_StringBuilder__ctor_System_Int32_) and a simple `for (int i = 0; i < shaBuffer.Length; i++)` loop.

Finally, please add a simple unit test for this method.